lib/ansible/modules/windows/setup.ps1

```diff
@@ -318,11 +318,12 @@ if($gather_subset.Contains('platform')) {
 
     $ansible_facts += @{
         ansible_architecture = $win32_os.OSArchitecture
-        ansible_domain = $ip_props.DomainName
-        ansible_fqdn = ($ip_props.Hostname + "." + $ip_props.DomainName)
-        ansible_hostname = $env:COMPUTERNAME
+        ansible_domain = $win32_cs.Domain.Substring($win32_cs.Workgroup.length)
```

Comment:
What's the purpose of the Substring here?

Reply:
A trick to get it to give empty string in a work group and the domain name in domain
to make sure the change backward compatible

Reply:
In my testing `Domain` is not set for a workgroup host, this means that this does nothing for those scenarios. Are you seeing something else here that makes this trick necessary?

Reply:
`Get-CimInstance win32_computersystem -Property domain,Workgroup | select domain,Workgroup`
On my laptop this gives the workgroup name in both, so if this were to run as is without the trick I would get a domain name, but there is no domain